FT2-37-1: Completion of Assignment 1.

Dutta-Sayan opened the pull request:
1. Consists  of 3 input fields, first name, last name and full name.
2. By default the full name field is disabled.
3. User can enter only alphabets in first name and last name.
4. On successful input, the full name field shows the name of the user along with a hello message.

Dutta-Sayan committed:
FT2-37-1: Assignment 1 Completed.
Description: 2 input fields created for first and last name containing only alphabets. The full name shows on succesful submission along with a greeting message.
Relevant styling has also been added to the website.

Dutta-Sayan committed:
FT2-37-1: Removed empty Readme file.

Dutta-Sayan committed:
FT2-37-1: Added further checks for validating user input.
Description: Confined the length of input.
Handled blankspace entry in the input field.

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+<?php
+    if(isset($_POST["submit"])) {
+        //fname, lname values hold the first and last name.
```
We should start commenting with capital letter with one space.

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+<?php
+    if(isset($_POST["submit"])) {
```
We should add a space after `if` statment.

kumareshinnoraft replied:
Add a one line space after starting `<?php` code.

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+        //fname, lname values hold the first and last name.
+        $fname= trim($_POST['fname']," ");
+        $lname= trim($_POST['lname'], " ");
+        //New object created.
```
We should start commenting with capital letter with one space.

`// New object created.`

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+        //New object created.
+        $user = new User($fname,$lname);
+
+        //returnMsg stores the value determining if invalid entry is done in input fields.
```
We should start commenting with capital letter with one space.

`// New object created.`

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+        $fname= trim($_POST['fname']," ");
+        $lname= trim($_POST['lname'], " ");
+        //New object created.
+        $user = new User($fname,$lname);
```
We should follow proper Indentation in the code.

`$user = new User($fname, $lname);`

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+<?php
+    if(isset($_POST["submit"])) {
+        //fname, lname values hold the first and last name.
+        $fname= trim($_POST['fname']," ");
```
We should add space before `=`.

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+        </div>
+    </section>
+</body>
+</html>
```
We should add an empty line at the end of the file.

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+            color: red;
+            font-size: 14px;
+            background-color: white;
+            /* padding: 5px; */
```
We should not push commented code in the PR.

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+            }
+        }
+        .greetings{
+            color: yellow;
```
Why we are not moving this CSS to different file?

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+            <h2>USER DETAILS</h2>
+            <form action="<?php echo $_SERVER["PHP_SELF"];?>" method="post" class="input-form">
+                First Name: <input type="text" name="fname" placeholder="First Name" value="<?php echo $fname?>" maxlength=25><br>
+                <span class="error"><?php if($returnMsg==1)echo $ferrMsg;?></span><br>
```
`<?php if($returnMsg==1)echo $ferrMsg;?`

Follow proper indentation in the code.

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+        and returns the message accordingly.
+        */
+        public function isValid() {
+            // $pattern = "/^[a-zA-Z]+[ ]*[a-zA-Z]+$/";
```
We should avoid sending commented code in the PR.

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+            // $pattern = "/^[a-zA-Z]+[ ]*[a-zA-Z]+$/";
+            $pattern = "/^[a-zA-Z]+$/";
+            if (!preg_match($pattern, $this->fname))
+                return 1;
```
Why we are using 1, 2 for the flag values?

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+            $this->lname = $lname;
+        }
+
+        /*
+        Checks if the input name follows the general naming pattern,
+        and returns the message accordingly.
+        */
```
This is not the proper way to write multiline code.

```
  /**
   * Checks if the input name follows the general naming pattern.
   *
   * @param type
   *
   * @return type
   */
```

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+            else if (!preg_match($pattern, $this->lname))
+                return 2;
+            else {
+                $greetings = "Hello $this->fname $this->lname";
```
Please, make `Hello` inside HTML tag so that FE team could add CSS.

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+                Last Name: <input type="text" name="lname" placeholder="Last Name" value="<?php echo $lname?>" maxlength=25><br>
+                <span class="error"><?php if($returnMsg==2)echo $lerrMsg;?></span><br>
+
+                Full Name: <input type="text" name="fullName"placeholder="Full Name" value ="<?php if($greetings!="")echo $fname." ".$lname?>"disabled><br>
```
`<?php if($greetings!="")echo $fname." ".$lname`

We should follow proper Indentation.

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+                <span class="error"><?php if($returnMsg==1)echo $ferrMsg;?></span><br>
+
+                Last Name: <input type="text" name="lname" placeholder="Last Name" value="<?php echo $lname?>" maxlength=25><br>
+                <span class="error"><?php if($returnMsg==2)echo $lerrMsg;?></span><br>
```
`<?php if($returnMsg==2)echo $lerrMsg;?`

We should follow proper Indentation.

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+        else
+            $greetings= $returnMsg;
+    }
+    class User {
```
Could we use create a separate file for this class?

kumareshinnoraft replied:
Please add class comments that must be in multiline.

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+            $greetings= $returnMsg;
+    }
+    class User {
+        public $fname, $lname;
```
We should add multiline comments for class variable.

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+    }
+    class User {
+        public $fname, $lname;
+        public function __construct($fname, $lname) {
+            $this->fname = $fname;
+            $this->lname = $lname;
+        }
```
We should add multiline comments for functions.

kumareshinnoraft commented on Assign1/index.php:
```diff
@@ -0,0 +1,117 @@
+    }
+    class User {
+        public $fname, $lname;
+        public function __construct($fname, $lname) {
```
We should add parameters for the variables.

kumareshinnoraft reviewed (changes requested):
1. PR title should be in one line.
2. Follow proper Indentation of the code.

Dutta-Sayan committed:
FT2-37-1: Made changes as requested in PR review.
Corrected the code indentation.
Added comments for functions explaining their working.
Applied check to disallow fullname field modification.
Made seperate files for css and User.php class.

Dutta-Sayan committed:
FT2-37-1: Added frontend validation of pattern checking of user input and handling empty input.

kunal-sin commented on Assign1/index.php:
```diff
@@ -0,0 +1,69 @@
+    <section class="user-details">
+        <div class="container">
+            <h2>USER DETAILS</h2>
+            <form action="<?php echo $_SERVER["PHP_SELF"];?>" method="post" class="input-form">
+                First Name: <input type="text" name="fname" placeholder="First Name" value="<?php echo $fname?>" maxlength=25 pattern="^[a-zA-Z]+$" required><br>
+                <span class="error">
+                    <?php if ($returnMsg == $fname) echo $ferrMsg; ?>
+                </span><br>
+
+                Last Name: <input type="text" name="lname" placeholder="Last Name" value="<?php echo $lname?>" maxlength=25 pattern="^[a-zA-Z]+$" required><br>
+                <span class="error">
+                    <?php if ($returnMsg == $lname) echo $lerrMsg; ?>
+                </span><br>
+
+                Full Name: <input type="text" name="fullName" placeholder="Full Name" value ="<?php if ($greetings != "") echo $fname." ".$lname ?>" disabled><br>
+                <span class="error">
+                    <?php if (!empty($err)) echo $err; ?>
+                </span><br>
+                <input  class="submit-button" type="submit" name="submit" value="Submit"> 
+            </form>
+            <?php if (!empty($greetings)) { ?>
+            <div class="greetings">Hello
```
Why we are adding php logical statements inside HTML?

kunal-sin replied:
It should be either above or below the HTML

Dutta-Sayan committed:
FT2-37-1: Removed PHP logic code from within HTML tags.

Dutta-Sayan committed:
FT2-37-1: Edited some comments of the class and function.

Dutta-Sayan committed:
FT2-37-1: Added frontend validation for the user name input field.

Dutta-Sayan committed:
FT2-37-1: Added pattern check and required option in the first name field.

Dutta-Sayan committed:
FT2-37-1: Added backend validation for maximum characters check and allowing space between names.
